### ISSUE TITLE
ERROR messages when running build_docs.py

### ISSUE THREAD

**Issue by kidonglee** [lowRISC/opentitan#1743]
Dear guys, 

I have trouble with ERROR messages when building local document server.

Right after cloning git, I followed the below sequences, as described in this repository.

```
$ sudo apt install curl python3 python3-pip
$ pip3 install --user -r python-requirements.txt
$ ./util/build_docs.py --preview
```

But, I got many ERROR messages as below,
```
....
2020-03-13 18:44 - Calling curl -sL https://github.com/gohugoio/hugo/releases/download/v0.60.0/hugo_extended_0.60.0_Linux-64bit.tar.gz | tar -xzO --overwrite hugo > /temp/opentitan_doc/opentitan/build/docs-hugo/hugo to download hugo.
Building sites … ERROR 2020/03/13 18:49:55 Symbolic links for directories not supported, skipping '/temp/opentitan_doc/opentitan/site/docs/layouts'
ERROR 2020/03/13 18:49:55 Failed to add template "partials/head.html" in path "/temp/opentitan_doc/opentitan/site/docs/layouts/partials/head.html": template: partials/head.html:27: function "resources" not defined
ERROR 2020/03/13 18:49:55 partials/head.html : template: partials/head.html:27: function "resources" not defined
ERROR 2020/03/13 18:49:55 hw/ip/i2c/doc/_index.md:102: unrecognized character in shortcode action: U+002E '.'. Note: Parameters with non-alphanumeric args must be quoted
ERROR 2020/03/13 18:49:55 Error while rendering "page" in "": template: _default/single.html:2:5: executing "_default/single.html" at <partial "head.html" ...>: error calling partial: template: "partials/head.html" is an incomplete or empty template
ERROR 2020/03/13 18:49:55 Error while rendering "page" in "doc/project/": template: _default/single.html:2:5: executing "_default/single.html" at <partial "head.html" ...>: error calling partial: template: "partials/head.html" is an incomplete or empty template
ERROR 2020/03/13 18:49:55 Error while rendering "page" in "doc/ug/": template: _default/single.html:2:5: executing "_default/single.html" at <partial "head.html" ...>: error calling partial: template: "partials/head.html" is an incomplete or empty template
....
```

If I remove 2 symbolic linked files, the 1st error by is disappeared.
However, I don't know how to fix the remaining ERRORs.
Please help.

Thanks in advance.

(Edit imphil: fix formatting)

**Comment by imphil** [comment]
Something seems to be strange with your setup. 

- What Linux distribution are you using?
- Did you clone your OpenTitan repository into `/temp/opentitan_doc/opentitan`? Does this path contain any symlinks, is it on a shared drive, or mounted into a virtual machine, etc?

**Comment by kidonglee** [comment]
Dear, imphil, 

Here is my answer.
1. My linux is Ubuntu 18.04 Desktop.
2. I make the directory named 'opentitan_doc' and then clone git, as follow,
    mkdir opentitan_doc
    cd opentitan_doc
    git clone https://github.com/lowRISC/opentitan.git
    ./util/build_docs.py --preview   // ====> make ERRORs
3. right after cloning git, I found 2 symblic link files, as below,
    ./hw/ip/rv_core_ibex/lint/rv_core_ibex.vlt -> ../../../vendor/lowrisc_ibex/lint/verilator_waiver.vlt
    ./site/docs/layouts/_default/list.html -> single.html

Thank you.

**Comment by imphil** [comment]
- The two symlinks you noticed are expected
- In my tests I cannot see any of the output or error messages you're seeing.  That's how it looks for me:


```
2020-03-13 15:00 - ['list of group', 'comma separated group of key:value enclosed in `{}` the second entry of the list is the sub group format']
2020-03-13 15:00 - Hugo not found. Installing local copy and re-trying.
2020-03-13 15:00 - Calling curl -sL https://github.com/gohugoio/hugo/releases/download/v0.60.0/hugo_extended_0.60.0_Linux-64bit.tar.gz | tar -xzO --overwrite hugo > /home/philipp/src/opentitan/build/docs-hugo/hugo to download hugo.

                   | EN   
+------------------+-----+
  Pages            | 237  
  Paginator pages  |   0  
  Non-page files   | 370  
  Static files     |   0  
  Processed images |   0  
  Aliases          |   0  
  Sitemaps         |   0  
  Cleaned          |   0  

Built in 433 ms
Watching for changes in /home/philipp/src/opentitan/{.Xil,.github,.pytest_cache,build,build-bin,build-out,ci,doc,hw,opentitan-docs,resources,scratch,site,sw,test,util}
Watching for config changes in /home/philipp/src/opentitan/site/docs/config.toml
Environment: "development"
Serving pages from /home/philipp/src/opentitan/build/docs
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
Press Ctrl+C to stop
```


Going back to my earlier questions:
- What kind of device is `/temp`? Ensure that you have no symlinks in your path somewhere.
- Are you sure that you're using the latest master version of opentitan?
- Please remove the clone and start from scratch to ensure that nothing is messing up your environment.

**Comment by kidonglee** [comment]
I appreciate for your response.
There is no symbolic link above "opentitan_doc" folder.
The "/temp" is mounted to the root (/) and the symbolic link related ERROR is disappeared after removing tho symbolic links that I mentioned above.
I will try to remove clone and start it again, as you recommend.
Thanks, again.

**Comment by kidonglee** [comment]
I fixed the above ERRORs by installing hugo extended version.
After then, I got another Error message, as below,
(FYI, actually, to fix the above ERROR messages, I tried to change folder installing opentitan. But it was not effective)
`Error: Error building site: "/tools/opentitan/opentitan/sw/vendor/google_googletest/googlemock/docs/pump_manual.md:1:1": plain HTML documents not supported`
I solve this error by removing the first line of "pump_manual.md".
Finally, I can run the local document server.
Thank you very much.
